src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/ApiController.cs

```diff
@@ -35,23 +36,29 @@ public class ApiController : Controller
                 return this.ChallengeOrForbid();
             }
 
-            var luceneQuery = new LuceneQuery
-            {
-                Index = indexName,
-                Template = query,
-                ReturnContentItems = true
-            };
+            var result = await LuceneQueryApi(indexName, query, parameters, returnContentItems: true);
 
-            var queryParameters = parameters != null ?
-                JsonConvert.DeserializeObject<Dictionary<string, object>>(parameters)
-                : new Dictionary<string, object>();
+            return new ObjectResult(result);
+        }
+        [HttpPost]
+        [Route("content")]
+        public async Task<IActionResult> ContentPost(
+            [FromForm] string indexName,
```

Comment:
What is we want to post as json?

Is it possible (am not sure, but I think this would work) to make this a `ViewModel` without the `[FromForm]` such that it would bind from both `Form` and `Json` posts?

Reply:
I thought about it,but then there are differences when using [ApiController] attribute the default source binder for complex type inference is [FormBody] and since the default media formatter is Json it will through an exception not supported formatter unless you use json body with application/json content type .Besides the query parameter value has liqued syntax which i think will require the client to escape or prefill the values of the parameters in the query
I am not sure as well ,maybe if we need custom processing.
I was thinking about it when i thought could we have a form submission request inside the workflow that read these form values and then  run a lucene query task and inspect the json result after.

Reply:
Hmm, I would still think this makes more sense as a json body than a form, for an api controller.

The parameters object must be supplied as json already in the get request, but it doesn't accept liquid syntax. Maybe you have misunderstood how the parameters object works here. It supplies the parameters, which are then read into the queries template context, and the template is then parsed as liquid using those parameter values.

It seems as unexpected as the post request only accepting query strings, that an api request would take `Form` values, (that must contain json in the parameters value).

Do you have a requirement to send form values to a lucene controller, rather than sending a json body? 

I'm just trying to understand the use case here?

> I was thinking about it when i thought could we have a form submission request inside the workflow that read these form values and then run a lucene query task and inspect the json result after.

I'm not sure how workflows connects with this api controller? Or maybe I have misunderstood the comment

Reply:
You are right, json is more suitable for the api controller especially for multi-client dev.
problem is , i don't know how would i deal with the query template parameter as json string since it has these {{ }} the controller binder will through an exception when trying to deserialize the value from the body.
Do you mean to send the query template parameter as json object becuase this would mean dealing with DSL of searching

Reply:
Workflow is a different thought sorry i mixed it here.

Reply:
Ok, cool, lets figure it out.

I'm not sure exactly, I would imagine trying to send the template parameter as a string value on a json object (not trying to deal with the DSL as a model).

You're right it might need some escaping (`JSON.stringify()`), or possibly a `JsonConvertor` that could handle the property as a string, rather than trying to treat it as a jobject. Hopefully it is simpler than that though.

Shall we get something that doesn't work, and see what that looks like, and see if I can help you from there? 

We should have a look and see what elastic search is doing here as well